### DISCUSSION TITLE
New Recrod entry error with table that has non-numeric key i.e varchar(16) or char(2)

### DISCUSSION THREAD

**Discussion by hafizTahir** [serenity-is/Serenity#5492] [General]
I have table :

CREATE TABLE [dbo].[faCategory](
	[Code] [char](2) NOT NULL,
	[Name] [char](40) NOT NULL
 CONSTRAINT [PK_faCategory] PRIMARY KEY CLUSTERED 
(
	[Code] ASC
)WITH (PAD_INDEX  = OFF, STATISTICS_NORECOMPUTE  = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS  = ON, ALLOW_PAGE_LOCKS  = ON) ON [PRIMARY]
) ON [PRIMARY]

The Code field is primary key i am generating it 01,02,03....99 in other software without serenity
but in serenity its giving me error.
How can i generate custom value for primary key in serenity 

Please help

**Comment by VictorTomaili** [comment]
https://serenity.is/demo/BasicSamples/ViewWithoutID

Also you can add an autoincrement id field to your table. It's will help you to handle it easy way if you have not strict reason.

**Reply by hafizTahir** [reply]
for that i have to change all those fields too where its using as foreign key.
Moreover, i have other tables that have master detail relation and using custom generated primary keys i.e
00001/0121
00002/0121 
etc.

**Reply by VictorTomaili** [reply]
you don't need change foreign key relations. You just create an unique autoincrement id field (not required as primary key) then use this id field as identity and id property on your row.cs will enough.

**Reply by hafizTahir** [reply]
i don't want the user to enter the next generated primary key for example the last record is 31 the next must be 32 and its datatype is char(2) .Is there any way to generate custom primary key by your own ?

**Reply by VictorTomaili** [reply]
try to read docs and check the demo pages before ask.
https://serenity.is/demo/BasicSamples/SerialAutoNumber

**Reply by hafizTahir** [reply]
when i try to access primary key column in CategoryDialog.ts its not accessible. I am new at Serenity Please guid me to generate new custom value for primary key column while inserting new records. i.e "01","02","03" in my case.


Thanks
